### PR TITLE
Update docker version of install script

### PR DESCRIPTION
**Pull request by martyav**
## Reminders

- See the [README](../README.md) for more details on how to work with the Rancher docs.

- Verify if changes pertain to other versions of Rancher. If they do, finalize the edits on one version of the page, then apply the edits to the other versions.

- If the pull request is dependent on an upcoming release, make sure to target the release branch instead of `main`.

## Description

From Slack:

> Since Docker 20.10.x would be [EOL next month](https://endoflife.date/docker-engine#:~:text=09%20Dec%202020)-,Ends%20in%201%20month,-(10%20Dec%202023). The rancher install script in the [documentation](https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/installation-requirements/install-docker#:~:text=curl%20https%3A//releases.rancher.com/install%2Ddocker/20.10.sh%20%7C%20sh) should be updated to latest docker install script (curl https://releases.rancher.com/install-docker/24.0.sh | sh)

## Comments

Rancher v2.8 can use the install script for Docker 24.0. However, earlier versions of Rancher are validated only up through 20.0, including certain setups for Rancher v2.7 (example -- Rancher v2.7 on SUSE Linux Enterprise Server). See The support matrix page: https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-7-9/

### CODE REVIEW

**Review by btat — CHANGES_REQUESTED** [review]
Your branch is outdated, can you rebase it.

We should also include a blurb asking users to consult the support matrix to see which version(s) are available for their OS. With that in place, we should take the same approach as other parts of our docs and not use a specific version number in the command.

**Comment by martyav** [comment]
The 5 commits added an hour ago represent an attempt to rebase to fix the outdated branch. I'm not very familiar with the rebasing operation so I'm not sure if that resolves that problem.

File: docs/getting-started/installation-and-upgrade/installation-requirements/install-docker.md [hunk 1]
```diff
@@ -10,18 +10,18 @@ Docker is required to be installed on nodes where the Rancher server will be ins
...
+For example, this command could be used to install on one of the main Linux distributions, such as SUSE Linux Enterprise or Ubuntu:
 
-```
-curl https://releases.rancher.com/install-docker/20.10.sh | sh
+```bash
+curl https://releases.rancher.com/install-docker/<version-number>.sh | sh
 ```
 
-Rancher has installation scripts for every version of upstream Docker that Kubernetes supports. To find out whether a script is available for installing a certain Docker version, refer to this [GitHub repository,](https://github.com/rancher/install-docker) which contains all of Rancher's Docker installation scripts.
+Consult the [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-7-9/) to match a validated Docker version with your operating system and version of Rancher. To find out whether a script is available for installing a certain Docker version, refer to this [GitHub repository,](https://github.com/rancher/install-docker) which contains all of Rancher's Docker installation scripts.
```
Comment (btat) [inline docs/getting-started/installation-and-upgrade/installation-requirements/install-docker.md:21]:
```suggestion
Consult the [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix) to match a validated Docker version with your operating system and version of Rancher. To find out whether a script is available for installing a certain Docker version, refer to this [GitHub repository,](https://github.com/rancher/install-docker) which contains all of Rancher's Docker installation scripts.
```

We should use either the unversioned link or hard code the 2.8 link, both of which will redirect to the latest live link right now. Harding coding 2.7 right now means we'll need to update this again shortly.

**Review by btat — CHANGES_REQUESTED** [review]
The version numbers in the support matrix list Major.Minor.Patch, but only Major.Minor matters for the script so that needs to be explained. Maybe with a text example?

File: docs/getting-started/installation-and-upgrade/installation-requirements/install-docker.md [hunk 2]
```diff
@@ -10,18 +10,20 @@ Docker is required to be installed on nodes where the Rancher server will be ins
...
+```bash
+curl https://releases.rancher.com/install-docker/<version-number>.sh | sh
 ```
-curl https://releases.rancher.com/install-docker/20.10.sh | sh
-```
 
-Rancher has installation scripts for every version of upstream Docker that Kubernetes supports. To find out whether a script is available for installing a certain Docker version, refer to this [GitHub repository,](https://github.com/rancher/install-docker) which contains all of Rancher's Docker installation scripts.
+Consult the [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix) to match a validated Docker version with your operating system and version of Rancher. Although the support matrix lists Rancher releases down to the patch version, only the major and minor version of the release are relevant for the Rancher installation scripts.
+
+To find out whether a script is available for installing a certain Docker version, refer to this [GitHub repository,](https://github.com/rancher/install-docker) which contains all of Rancher's Docker installation scripts.
```
Comment (btat) [inline docs/getting-started/installation-and-upgrade/installation-requirements/install-docker.md:23]:
```suggestion
Consult the [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix) to match a validated Docker version with your operating system and version of Rancher. Although the support matrix lists validated Docker versions down to the patch version, only the major and minor version of the release are relevant for the Docker installation scripts.
```

File: versioned_docs/version-2.0-2.4/getting-started/installation-and-upgrade/installation-requirements/install-docker.md [hunk 3]
```diff
@@ -10,12 +10,14 @@ Docker is required to be installed on nodes where the Rancher server will be ins
...
+```bash
+curl https://releases.rancher.com/install-docker/<version-number>.sh | sh
 ```
-curl https://releases.rancher.com/install-docker/19.03.sh | sh
-```
 
-Rancher has installation scripts for every version of upstream Docker that Kubernetes supports. To find out whether a script is available for installing a certain Docker version, refer to this [GitHub repository,](https://github.com/rancher/install-docker) which contains all of Rancher's Docker installation scripts.
+Consult the [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-4-18/) to match a validated Docker version with your operating system and version of Rancher. Although the support matrix lists Rancher releases down to the patch version, only the major and minor version of the release are relevant for the Rancher installation scripts.
+
+To find out whether a script is available for installing a certain Docker version, refer to this [GitHub repository,](https://github.com/rancher/install-docker) which contains all of Rancher's Docker installation scripts.
```
Comment (btat) [inline versioned_docs/version-2.0-2.4/getting-started/installation-and-upgrade/installation-requirements/install-docker.md:23]:
```suggestion
Consult the [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-4-18/) to match a validated Docker version with your operating system and version of Rancher. Although the support matrix lists validated Docker versions down to the patch version, only the major and minor version of the release are relevant for the Docker installation scripts.
```

File: versioned_docs/version-2.5/getting-started/installation-and-upgrade/installation-requirements/install-docker.md [hunk 4]
```diff
@@ -10,12 +10,14 @@ Docker is required to be installed on nodes where the Rancher server will be ins
...
+```bash
+curl https://releases.rancher.com/install-docker/<version-number>.sh | sh
 ```
-curl https://releases.rancher.com/install-docker/20.10.sh | sh
-```
 
-Rancher has installation scripts for every version of upstream Docker that Kubernetes supports. To find out whether a script is available for installing a certain Docker version, refer to this [GitHub repository,](https://github.com/rancher/install-docker) which contains all of Rancher's Docker installation scripts.
+Consult the [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-5-16/) to match a validated Docker version with your operating system and version of Rancher. Although the support matrix lists Rancher releases down to the patch version, only the major and minor version of the release are relevant for the Rancher installation scripts.
+
+To find out whether a script is available for installing a certain Docker version, refer to this [GitHub repository,](https://github.com/rancher/install-docker) which contains all of Rancher's Docker installation scripts.
```
Comment (btat) [inline versioned_docs/version-2.5/getting-started/installation-and-upgrade/installation-requirements/install-docker.md:23]:
```suggestion
Consult the [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-5-16/) to match a validated Docker version with your operating system and version of Rancher. Although the support matrix lists validated Docker versions down to the patch version, only the major and minor version of the release are relevant for the Docker installation scripts.
```

File: versioned_docs/version-2.6/getting-started/installation-and-upgrade/installation-requirements/install-docker.md [hunk 5]
```diff
@@ -10,15 +10,17 @@ Docker is required to be installed on nodes where the Rancher server will be ins
...
+```bash
+curl https://releases.rancher.com/install-docker/<version-number>.sh | sh
 ```
-curl https://releases.rancher.com/install-docker/20.10.sh | sh
-```
 
-Rancher has installation scripts for every version of upstream Docker that Kubernetes supports. To find out whether a script is available for installing a certain Docker version, refer to this [GitHub repository,](https://github.com/rancher/install-docker) which contains all of Rancher's Docker installation scripts.
+Consult the [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-6-13/) to match a validated Docker version with your operating system and version of Rancher. Although the support matrix lists Rancher releases down to the patch version, only the major and minor version of the release are relevant for the Rancher installation scripts.
+
+To find out whether a script is available for installing a certain Docker version, refer to this [GitHub repository,](https://github.com/rancher/install-docker) which contains all of Rancher's Docker installation scripts.
```
Comment (btat) [inline versioned_docs/version-2.6/getting-started/installation-and-upgrade/installation-requirements/install-docker.md:23]:
```suggestion
Consult the [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-6-13/) to match a validated Docker version with your operating system and version of Rancher. Although the support matrix lists validated Docker versions down to the patch version, only the major and minor version of the release are relevant for the Docker installation scripts.
```

File: versioned_docs/version-2.7/getting-started/installation-and-upgrade/installation-requirements/install-docker.md [hunk 6]
```diff
@@ -10,18 +10,20 @@ Docker is required to be installed on nodes where the Rancher server will be ins
...
+```bash
+curl https://releases.rancher.com/install-docker/<version-number>.sh | sh
 ```
-curl https://releases.rancher.com/install-docker/20.10.sh | sh
-```
 
-Rancher has installation scripts for every version of upstream Docker that Kubernetes supports. To find out whether a script is available for installing a certain Docker version, refer to this [GitHub repository,](https://github.com/rancher/install-docker) which contains all of Rancher's Docker installation scripts.
+Consult the [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-7-9/) to match a validated Docker version with your operating system and version of Rancher. Although the support matrix lists Rancher releases down to the patch version, only the major and minor version of the release are relevant for the Rancher installation scripts.
+
+To find out whether a script is available for installing a certain Docker version, refer to this [GitHub repository,](https://github.com/rancher/install-docker) which contains all of Rancher's Docker installation scripts.
```
Comment (btat) [inline versioned_docs/version-2.7/getting-started/installation-and-upgrade/installation-requirements/install-docker.md:23]:
```suggestion
Consult the [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-7-9/) to match a validated Docker version with your operating system and version of Rancher. Although the support matrix lists validated Docker versions down to the patch version, only the major and minor version of the release are relevant for the Docker installation scripts.
```

File: versioned_docs/version-2.8/getting-started/installation-and-upgrade/installation-requirements/install-docker.md [hunk 7]
```diff
@@ -10,18 +10,20 @@ Docker is required to be installed on nodes where the Rancher server will be ins
...
+```bash
+curl https://releases.rancher.com/install-docker/<version-number>.sh | sh
 ```
-curl https://releases.rancher.com/install-docker/20.10.sh | sh
-```
 
-Rancher has installation scripts for every version of upstream Docker that Kubernetes supports. To find out whether a script is available for installing a certain Docker version, refer to this [GitHub repository,](https://github.com/rancher/install-docker) which contains all of Rancher's Docker installation scripts.
+Consult the [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-8-0/) to match a validated Docker version with your operating system and version of Rancher. Although the support matrix lists Rancher releases down to the patch version, only the major and minor version of the release are relevant for the Rancher installation scripts.
+
+To find out whether a script is available for installing a certain Docker version, refer to this [GitHub repository,](https://github.com/rancher/install-docker) which contains all of Rancher's Docker installation scripts.
```
Comment (btat) [inline versioned_docs/version-2.8/getting-started/installation-and-upgrade/installation-requirements/install-docker.md:23]:
```suggestion
Consult the [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-8-0/) to match a validated Docker version with your operating system and version of Rancher. Although the support matrix lists validated Docker versions down to the patch version, only the major and minor version of the release are relevant for the Docker installation scripts.
```

**Review by btat — CHANGES_REQUESTED** [review]

**Review by btat — APPROVED** [review]

**Pull request merged**